lib/ansible/executor/module_common.py

```diff
@@ -890,3 +918,16 @@ def modify_module(module_name, module_path, module_args, templar, task_vars=None
         b_module_data = b"\n".join(b_lines)
 
     return (b_module_data, module_style, shebang)
+
+
+class InterpreterDiscoveryRequiredError(Exception):
+    def __init__(self, message, interpreter_name, discovery_mode):
+        super(InterpreterDiscoveryRequiredError, self).__init__(message)
+        self.interpreter_name = interpreter_name
+        self.discovery_mode = discovery_mode
+
+    def __str__(self):
+        return self.message
+
+    def __repr__(self):
+        return self.message
```

Comment:
repr should say how to recreate the class whenever possible:
```
>>> repr(err)
"ZeroDivisionError('integer division or modulo by zero',)"
>>> print(err)
integer division or modulo by zero
```
```suggestion
        return 'InterpreterDiscoveryRequiredError({0}, {1}, {2})'.format(self.message, self.interpreter_name, self.discover_mode)
```